Replace zenity select dropdown with a list

dragunoff opened the pull request:
Change the zenity select dropdown to a list. The list has better keyboard usability because it is navigable with up/down arrows and pressing Enter submits the selected item. It is also searchable so you can press "D" to get straight to d2k. Unfortunately I couldn't find a way to make the first item selected by default. A list is better than a radio list because the radio list doesn't respond to any keyboard input (except Tab). I also added a title to the window "Launch OpenRA".

![Screenshot_20210927_185027](https://user-images.githubusercontent.com/1355810/134942631-0748a223-4154-4dd3-b16d-0af0c7af309a.png)
(items overflow because my display is scaled at 125%, it should be fine on 100%)

Supersedes #19463

dragunoff committed:
Replace zenity select dropdown with a list

anvilvapre commented:
Optionally consider adding `shellcheck` to the `Makefile` to validate shell scripts. Or to run it once.

It helps in keeping your scripts cross platform compatible.

pchote commented:
This is a great polish win.

Before:
<img width="336" alt="Screenshot 2021-09-27 at 21 29 15" src="https://user-images.githubusercontent.com/167819/134980542-9dfd02be-5af8-4d0e-ab62-1b8f5d42a4b8.png">

After:
<img width="403" alt="Screenshot 2021-09-27 at 21 28 58" src="https://user-images.githubusercontent.com/167819/134980550-851d5c56-87d0-4d88-b93e-19facc9f342b.png">

One small oversight and one small scope creep:
* `--list` is specified twice in the command.
* Can we also change the `if [ z"${*#*Game.Mod}" = z"$*" ]` to `if [ z"${*#*Game.Mod=}" = z"$*" ]` so it doesn't falsely match on arguments like `Game.ModeratelyAnnoyingFalsePositive=oops`?

Otherwise LGTM (and I tested that this still works after amending my suggestions).

dragunoff committed:
Improve Game.Mod argument check in launch script

dragunoff commented:
Update: addressed comments.

I ran shellcheck and it raised two warnings:
```
In launch-game.sh line 16:
if [ z"${*#*Game.Mod=}" = z"$*" ]
       ^-------------^ SC2039: In POSIX sh, string operations on $@/$* are undefined.


In launch-game.sh line 37:
if [ $? != 0 ] && [ $? != 1 ]; then
     ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0b893ca36394e9cc7db2254489c5e14670c852fa...9b209c54915c2c458aa662bb1f4723a969f5afb9)
